src/mon/OSDMonitor.cc

```diff
@@ -5128,19 +5129,32 @@ bool OSDMonitor::handle_osd_timeouts(const utime_t &now,
     }
     if (!osdmap.is_up(i))
       continue;
-    const std::map<int,utime_t>::const_iterator t = last_osd_report.find(i);
+    const std::map<int, std::pair<utime_t, int>>::const_iterator t = last_osd_report.find(i);
     if (t == last_osd_report.end()) {
       // it wasn't in the map; start the timer.
-      last_osd_report[i] = now;
+      last_osd_report[i].first = now;
+      last_osd_report[i].second = g_conf()->osd_beacon_report_interval;
     } else if (can_mark_down(i)) {
-      utime_t diff = now - t->second;
+      utime_t diff = now - t->second.first;
       if (diff > timeo) {
-	mon->clog->info() << "osd." << i << " marked down after no beacon for "
-			  << diff << " seconds";
-	derr << "no beacon from osd." << i << " since " << t->second
-	     << ", " << diff << " seconds ago.  marking down" << dendl;
-	pending_inc.new_state[i] = CEPH_OSD_UP;
-	new_down = true;
+	// check mon_osd_report_timout and osd_beacon_report_interval when a timeout happens.
+	if (g_conf()->mon_osd_report_timeout >= 2 * t->second.second) {
```

Comment:
and this needs to also check if t->second.second != utime_t(), or else do nothing (so that older OSDs don't trigger the warning)